scripts/globals/beastmentreasure.lua

```diff
@@ -0,0 +1,394 @@
+------------------------------------
+-- Beastmen Treasure side quests
+-- https://ffxiclopedia.fandom.com/wiki/Beastmen_Treasure
+------------------------------------
+require('scripts/globals/keyitems')
+require('scripts/globals/quests')
+require('scripts/globals/status')
+require('scripts/globals/weather')
+------------------------------------
+dsp = dsp or {}
+dsp.beastmentreasure = dsp.beastmentreasure or {}
+
+local PICKAXE = 605
+
+local zoneData = {
+    -- Yuhtunga Jungle (Elshimo Lowlands)
+    [123] =
+    {
+        statusvar = 'BMT_Lowlands_Status',
+        qmvar = 'BMT_Lowlands_DigQM',
+        mapid = dsp.ki.MAP_OF_THE_ELSHIMO_REGIONS,
+        peddlestox =
+        {
+            id = 17281639,
+            day = dsp.day.WINDSDAY
+        },
+        fetchitems =
+        {
+            1480, -- Mermaid Head
+            1481, -- Mermaid Body
+            1482, -- Mermaid Hands
+            1483  -- Mermaid Tail
+        },
+        digsites =
+        {
+            [3]  = {id = 17281642, x = 83.726,   y = 448.329,  chest = 0},
+            [4]  = {id = 17281643, x = -114.254, y = -125.471, chest = 1},
+            [5]  = {id = 17281644, x = -300.823, y = 9.208,    chest = 2},
+            [6]  = {id = 17281645, x = -539.311, y = 203.151,  chest = 3},
+            [7]  = {id = 17281646, x = -577.734, y = -82.563,  chest = 4},
+            [8]  = {id = 17281647, x = -361.835, y = -399.517, chest = 5},
+            [9]  = {id = 17281648, x = -294.827, y = -453.237, chest = 6},
+            [10] = {id = 17281649, x = 369.795,  y = 201.805,  chest = 7}
+        },
+        loot =
+        {
+            unique = {[14626] = 5}, -- Mermaid's Ring
```

Comment:
I'm open to a different approach for loot, but this is what came closest to the data I collected. Part of the problem here is there's such a small set of reported data that it's hard to know how accurate these are. Not a lot of people posted full rewards to begin with, and almost no one has even talked about the Yhoator/W.Altep variants. It doesn't help that these quests are time-consuming and the rewards are almost entirely irrelevant.

Feel free to take a look at the ["reward reports" tab of this spreadsheet](https://docs.google.com/spreadsheets/d/1Xw4fZ3ixM2sLCMq57IH19oEblpi_TZt0gLtVXyKar48/edit#gid=347371406) if you haven't already, and let me know what you think. To me it looked like a roughly 2:1 chance to roll a tier 1 gem over a colored rock, and regional junk rolled way too often to be 1:1 in a pool with all seeds. Incidentally, the only colored rocks that are confirmed with any degree of confidence are red and blue, and the only seeds are vegetable and grain. It starts to look a lot more like a true random if we assume those are actually the only items in the pool, but that seems weird too.

Reply:
I think what retail does is the same thing as opening a BCNM's treasure box. I ran a lot of these back in the old days when I was active on retail a SEA was still the endgame, and unfortunately I never recorded what all I got nor do I exactly recall which drops came out. I was going for the uniques. Got my sand mantle on my 3rd try.

Reply:
Yeah, it definitely looks *a lot* like a BCNM crate. In fact, before I wrote the loot distribution I'd hoped to just re-use existing BCNM code for it, but didn't see an easy way to do that. With no accessible mechanism for this, my thought was to either modify the BCNM code (which I wasn't super comfortable with,) or design something similar that could potentially be used with other quests in the future, so I started out defining some of these functions globally. At some point I made it all local because it felt a bit "out of scope" (sorry,) and I couldn't think of any other quests that might actually use it. But it's the kind of thing that *could* be made public or extracted and expanded upon if there's a valid use case for it.

It might also be worth noting here that the wiki has a lot more items listed as possible rewards for these than what I landed on. They list the entire spectrum of colored chips and nuggets, for example, when there's literally one guy on Alla saying he got a red chip and a gold nugget, and nothing else on the internet to corroborate. It seems likely that SE had predefined pools available to both BCNMs and quests like this, but without a large enough data set to figure them out it's basically guess-work.

Reply:
I've def gotten chips before. Dunno about nuggets. probably has a "slot" that can turn up any color of chip like the colored rocks are together in their slot with gems. No idea what else would be with them though.